dev-guide/api-conventions.md

```diff
@@ -109,6 +109,16 @@ This means, that any end user fetching the resource from the API can observe tha
 This has the effect of helping our users to understand how they might be able to configure their cluster, without
 having to search for features or review API schemas within the product docs.
 
+##### Pointers to structs
+
+An exception to this rule is when using a pointer to a struct in combination with an API validation that requires the field to be unset.
+
+The JSON tag `omitempty` is not compatible with struct references. Meaning any struct will always, when empty, be marshalled to `{}`. If the struct **must** genuinely be omitted, it must be a pointer.
```

Comment:
under what condition must the struct be omitted?  Discriminated unions, yes. But anywhere else?  I would expect the zero value to be exactly the same as nil and missing.

Reply:
This looks great. Can you mention all of it in the document?

Reply:
Is there a specific part you think is particularly useful? I worry adding too many in depth examples will lead to the document being overwhelmingly long and complicated to read

Perhaps a link to this thread would make that easier?

Reply:
The link works as well.

Reply:
So while I understand that it makes the validation slightly easier, for a configuration resource, why shouldn't we expect a zero-value of rootVolume to behave the same as a nil rootVolume?

For a workload resource, I definitely agree it should be a pointer?

Reply:
One example would be a discriminated union. We use those on configuration resources regularly. The DU needs a required discriminator, but, you cannot have a required field in a non-pointer struct. So any DU must be a pointer to struct independent of configuration or not.

In other examples where we want a required field in a struct, say for example an optional reference to another object with a namespace. If you specify the reference, we want the name and namespace field within to be required. 

Is it better here to use CEL to check for "all required values are empty" and allow that case? Does that become a little confusing to the user?